### PR TITLE
Add Javascript SDK

### PR DESCRIPTION
**Pull request by tonilopezmr**
* Adds Javascript SDK initial setup
* Reserved seaplane npm package
* Adds support for CommonJS and Typescript
* Adds  lint, format, prettier (formatter), and husky (pre-commit hook)

[Update] Added later after the first review: 
* Adds CI workflows for test
* Adds Configuration class
* Adds Identify API endpoints
* Adds Metadata API endpoints
* Adds logs and configuration for logs
* Adds autoRenew functionality, etc.

You can run thee tests using: 

*  npm run tests

## Usage

```javascript
const config = new Configuration({ 
  apiKey: "sp-kXl8a0WsfhzrcwZ6hbiAKw"  
})
const metadata = new Metadata(config)

await metadata.set({key: "key", value: "value"})
await metadata.getPage()
await metadata.delete({key: "key"})
await metadata.getPage({directory: { key: "foo" }})
```

### CODE REVIEW

**Comment by tonilopezmr** [comment]
Current dependencies on/for this PR:
* main
  * **PR #268** <a href="https://app.graphite.dev/github/pr/seaplane-io/seaplane/268" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/seaplane-io/seaplane/268?utm_source=stack-comment).

File: seaplane-sdk/javascript/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+Licensed under the Apache License, Version 2.0, [LICENSE]. Copyright 2022 Seaplane IO, Inc.
+
+[//]: # (Links)
+
+[Seaplane]: https://seaplane.io/
+[CLI]: https://github.com/seaplane-io/seaplane/tree/main/seaplane-cli
+[SDK]: https://github.com/seaplane-io/seaplane/tree/main/seaplane
+[Getting Started]: https://github.com/seaplane-io/seaplane/blob/main/seaplane-sdk/python/docs/quickstart.md
+[CONTRIBUTING]: https://github.com/seaplane-io/seaplane/tree/main/seaplane-sdk/python/CONTRIBUTIONS.md
+[LICENSE]: https://github.com/seaplane-io/seaplane/blob/main/LICENSE
```
Comment (kbknapp) [inline seaplane-sdk/javascript/README.md:37]: We can remove all these unused links, no?

File: seaplane-sdk/javascript/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const world = 'world';
+
+export function hello(who: string = world): string {
+  return `Welcome ${who} to seaplane! `;
+}
```
Comment (kbknapp) [inline seaplane-sdk/javascript/src/index.ts:5]: I'm not usually a fan of committing hello-world stubs to a public repo, I'd prefer we flesh it out a little more before including or at a minimum just a an `error` saying it's under construction.

File: seaplane-sdk/javascript/tests/index.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+const seaplane = require("seaplane")
+
+result = seaplane.hello("Toni")
+console.log(result)
```
Comment (kbknapp) [inline seaplane-sdk/javascript/tests/index.js:4]: Same here

**Review by kbknapp — COMMENTED** [review]
Looks like good scaffolding. I'd prefer to have at least some stubbing or functionality before merging to main. If you'd prefer you can always work against a feature branch incrementally and them merge that to main when ready?

At a minimum we should probably change the license of the tests module.

**Comment by tonilopezmr** [comment]
I agree @kbknapp, I just wanted to upload everything in small steps to see progress, I'll continue working on this branch until I have the `Configuration` and `Seaplane` classes done.

Thanks for your review! 🎉

**Comment by kbknapp** [comment]
No worries! Just ping me again when you're ready and I'll give the 👍🏻

**Review by tiago-peres — CHANGES_REQUESTED** [review]
Good with me... I'd just address all of @kbknapp's comments before merging

**Review by tiago-peres — APPROVED** [review]
LGTM

**Comment by tonilopezmr** [comment]
This Linear [CUST-264](https://linear.app/seaplane/issue/CUST-264/metadata-api) was included here as well

**Pull request merged**